diagnostic_feedback/quiz.py

```diff
@@ -0,0 +1,260 @@
+import logging
+import copy
+from xblock.core import XBlock
+from xblock.fields import Scope, String, List, Integer, Dict
+from xblock.fragment import Fragment
+from xblockutils.resources import ResourceLoader
+from .mixins import ResourceMixin, XBlockWithTranslationServiceMixin
+from .quiz_result import QuizResultMixin
+from .helpers import MainHelper
+from .validators import Validator
+from .sub_api import my_api
+from .data_tool import ExportDataBlock
+
+log = logging.getLogger(__name__)
+loader = ResourceLoader(__name__)
+
+
+# Make '_' a no-op so we can scrape strings
+def _(text):
+    return text
+
+
+@XBlock.needs('i18n')
+@XBlock.wants('user')
+class QuizBlock(ResourceMixin, QuizResultMixin, ExportDataBlock, XBlockWithTranslationServiceMixin):
+    """
+    An XBlock which can be used to add diagnostic quiz
+    """
+    BUZZFEED_QUIZ_VALUE = "BFQ"
+    BUZZFEED_QUIZ_LABEL = _("BuzzFeed-style")
+    DIAGNOSTIC_QUIZ_VALUE = "DG"
+    DIAGNOSTIC_QUIZ_LABEL = _("Diagnostic-style")
+
+    display_name = String(
+        display_name=_("Diagnostic Feedback"),
+        help=_("This name appears in the horizontal navigation at the top of the page."),
+        scope=Scope.settings,
+        default=""
```

Comment:
Perhaps the default should be `_("Quiz")`? Or implement `@property def display_name_with_default(self):` and return the quiz title.

The blank default as used here gives a strange tooltip in the LMS:
![screen shot 2015-11-30 at 6 30 07 pm](https://cloud.githubusercontent.com/assets/945577/11490749/9c6ec1d8-9790-11e5-855b-9df460d639ee.png)